modules/apps/analytics/analytics-client-js/src/analytics.js

```diff
@@ -159,22 +160,28 @@ class Analytics {
 
 	/**
 	 * Gets the userId for the existing analytics user. Previously generated ids
-	 * are stored and retrieved before generating a new one and attempting to update
-	 * the Identity Service.
+	 * are stored and retrieved before generating a new one. If a anonymous 
+	 * navigation is started after a identified navigation, the user ID token
+	 * is regenerated.
 	 * @return {Promise} A promise resolved with the stored or generated userId
 	 */
 	_getUserId() {
-		let userId = storage.get(STORAGE_KEY_USER_ID);
+		const {anonymous} = this.config;
 
-		if (userId) {
-			return Promise.resolve(userId);
-		} else {
-			userId = this._generateUserId();
+		const storedAnonymous = storage.get(STORAGE_KEY_ANONYMOUS);
+		const storedUserId = storage.get(STORAGE_KEY_USER_ID);
+
+		if (!storedUserId || ((storedAnonymous === false) && (anonymous === true))) {
+			const userId = this._generateUserId();
 
+			this._persist(STORAGE_KEY_ANONYMOUS, anonymous);
 			this._persist(STORAGE_KEY_USER_ID, userId);
 
 			return Promise.resolve(userId);
 		}
+		else {
+			return Promise.resolve(userId);
```

Comment:
Is this right? I mean, the userId seems to not exist in the else scope

Reply:
You're right, it should be `storedUserId`, I forgot to update this variable.

Reply:
Fixed, 9e9ad934b1a2e603cc42e554ee0aed20967e558c